test/arrayops.jl

```diff
@@ -702,8 +702,17 @@ end
     @test_throws ArgumentError Base.circshift!(a, a, 1)
     b = copy(a)
     @test Base.circshift!(b, a, 1) == [5,1,2,3,4]
```

Comment:
You can just add `@inferred(...)` to this.

Reply:
Ok, I guess it can't hurt. But note that the tests I added before are needed.
Inference succeeds without this PR (the type assertion), for arrays of dimension 1 & 2. So this test would pass, even without this PR.